src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -73,9 +74,10 @@ public static SecurityStatusPal AcceptSecurityContext(ref SafeFreeCredentials cr
             return SecurityStatusAdapterPal.GetSecurityStatusPalFromNativeInt(errorCode);
         }
 
-        public static SecurityStatusPal InitializeSecurityContext(ref SafeFreeCredentials credentialsHandle, ref SafeDeleteSslContext context, string targetName, ArraySegment<byte> input, ref byte[] outputBuffer, SslAuthenticationOptions sslAuthenticationOptions)
+        public static SecurityStatusPal InitializeSecurityContext(ref SafeFreeCredentials credentialsHandle, ref SafeDeleteSslContext context, string targetName, byte[] inputBuffer, int offset, int count, ref byte[] outputBuffer, SslAuthenticationOptions sslAuthenticationOptions)
         {
             Interop.SspiCli.ContextFlags unusedAttributes = default;
+            ArraySegment<byte> input = inputBuffer != null ? new ArraySegment<byte>(inputBuffer, offset, count) : default;
```

Comment:
It seems pretty strange to have moved this ArraySegment build from the caller to the callees.  ArraySegments can go to spans just fine (input.AsSpan()).

Reply:
I originally started with Span much earlier. But lot of the inner windows code expect  ArraySegment. So I decided to push byte[] till platform split and and use Span/Memory for Unix and keep ArraySegment for Windows.  That can be updated later independently of the APM->Task change. For Unix, byte[] -> ArraySegment->Span seems less optimal than byte[] -> Span.